odl/tomo/operators/xray_trafo.py

```diff
@@ -0,0 +1,285 @@
+# Copyright 2014, 2015 The ODL development group
+#
+# This file is part of ODL.
+#
+# ODL is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# ODL is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with ODL.  If not, see <http://www.gnu.org/licenses/>.
+
+"""X-ray transforms."""
+
+# Imports for common Python 2/3 codebase
+from __future__ import print_function, division, absolute_import
+from future import standard_library
+standard_library.install_aliases()
+from future.builtins import str, super
+
+# External
+import numpy as np
+
+# Internal
+from odl.discr.lp_discr import DiscreteLp
+from odl.space import FunctionSpace
+from odl.operator.operator import Operator
+from odl.tomo.geometry.geometry import Geometry
+from odl.tomo.backends import (
+    ASTRA_AVAILABLE, ASTRA_CUDA_AVAILABLE,
+    astra_cpu_forward_projector, astra_cpu_back_projector,
+    astra_cuda_forward_projector, astra_cuda_back_projector)
+
+
+_SUPPORTED_BACKENDS = ('astra_cpu', 'astra_cuda')
+
+__all__ = ('XrayTransform', 'XrayBackProjector')
+
+
+# TODO: DivergentBeamTransform
```

Comment:
I once proposed that we split this operator into the parallel case (X-ray transform) and the divergent beam case (divergent beam transform). This is debatable. They would surely share a fair amount of code and they are both ray-based. Maybe a single guy called simply `RayTransform` would be a compromise?